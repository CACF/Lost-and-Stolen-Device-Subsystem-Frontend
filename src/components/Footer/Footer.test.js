import React from 'react';
import Footer from './Footer';
import {mount, shallow} from 'enzyme';
import { spy } from 'sinon'

describe('Footer Component', ()=> {

  test('if renders correctly', ()=>{
    const wrapper = shallow(<Footer />);
    expect(wrapper).toMatchSnapshot();
  });

  test('if Footer wrapper contains app-footer class', ()=>{
    const wrapper  = mount(<Footer />)
    expect(wrapper.find('footer').hasClass('app-footer')).toBe(true);
  })

  test('contains two divs', ()=>{
    const wrapper  = mount(<Footer />)
    expect(wrapper.find('div')).toHaveLength(2);
  })
})