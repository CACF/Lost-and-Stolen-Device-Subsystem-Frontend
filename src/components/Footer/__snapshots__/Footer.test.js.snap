// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component if renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          © 
          Copyright
           
          2019
           
          <i>
            DIRBS
          </i>
          . 
          All rights reserved
          .
        </div>,
        <div>
          <b>
            Version
            : 
          </b>
          2.0.0
        </div>,
      ],
      "className": "app-footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "© ",
            "Copyright",
            " ",
            2019,
            " ",
            <i>
              DIRBS
            </i>,
            ". ",
            "All rights reserved",
            ".",
          ],
        },
        "ref": null,
        "rendered": Array [
          "© ",
          "Copyright",
          " ",
          2019,
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "DIRBS",
            },
            "ref": null,
            "rendered": "DIRBS",
            "type": "i",
          },
          ". ",
          "All rights reserved",
          ".",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <b>
              Version
              : 
            </b>,
            "2.0.0",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Version",
                ": ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Version",
              ": ",
            ],
            "type": "b",
          },
          "2.0.0",
        ],
        "type": "div",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            © 
            Copyright
             
            2019
             
            <i>
              DIRBS
            </i>
            . 
            All rights reserved
            .
          </div>,
          <div>
            <b>
              Version
              : 
            </b>
            2.0.0
          </div>,
        ],
        "className": "app-footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "© ",
              "Copyright",
              " ",
              2019,
              " ",
              <i>
                DIRBS
              </i>,
              ". ",
              "All rights reserved",
              ".",
            ],
          },
          "ref": null,
          "rendered": Array [
            "© ",
            "Copyright",
            " ",
            2019,
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "DIRBS",
              },
              "ref": null,
              "rendered": "DIRBS",
              "type": "i",
            },
            ". ",
            "All rights reserved",
            ".",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <b>
                Version
                : 
              </b>,
              "2.0.0",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Version",
                  ": ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Version",
                ": ",
              ],
              "type": "b",
            },
            "2.0.0",
          ],
          "type": "div",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
